src/prefect/server/api/concurrency_limits.py

```diff
@@ -127,14 +247,48 @@ async def reset_concurrency_limit_by_tag(
     ),
     db: PrefectDBInterface = Depends(provide_database_interface),
 ) -> None:
+    # Reset V2 leases if V2 limit exists; otherwise fall back to V1
+    v2_name = f"tag:{tag}"
+
     async with db.session_context(begin_transaction=True) as session:
-        model = await models.concurrency_limits.reset_concurrency_limit_by_tag(
-            session=session, tag=tag, slot_override=slot_override
-        )
-    if not model:
-        raise HTTPException(
-            status_code=status.HTTP_404_NOT_FOUND, detail="Concurrency limit not found"
-        )
+        model = await cl_v2_models.read_concurrency_limit(session=session, name=v2_name)
+
+        if not model:
+            # Fall back to V1
+            model = await models.concurrency_limits.reset_concurrency_limit_by_tag(
+                session=session, tag=tag, slot_override=slot_override
+            )
+            if not model:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Concurrency limit not found",
+                )
+            return
+
+        # Revoke all existing leases for this limit
+        lease_storage = get_concurrency_lease_storage()
+        active_leases = await lease_storage.read_active_lease_ids(limit=1000)
```

Comment:
We should probably paginate through these instead of assuming there will always be less than 1000.